### PR TITLE
feat: create plugin architecture [RQ-0101]

### PR DESCRIPTION
**Pull request by sjavaheri**
🚀 feat: create plugin architecture [RQ-0101] 🚀 

Plugin architecture created under the folder MIND. Plugin can be compiled with the `compile_plugins.bat` script under `vaa3D_tools/released_plugins`

[Documentation ](https://docs.google.com/document/d/1lssBzgK_OXZN2cvymKzt4f81mYddF4tRpPnWMwNvsvs/edit?tab=t.0#heading=h.kvlkdx4al550)of how to create and compile plugins complete

### CODE REVIEW

File: released_plugins/v3d_plugins/MIND/soma_segmentation.pro [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+TEMPLATE	= lib
+CONFIG	+= qt plugin warn_off
+QT += widgets
+#CONFIG	+= x86_64
+VAA3DPATH = C:/Users/shida/Offline/Projects/vaa3d/v3d_external
```
Comment (sigerma88) [inline released_plugins/v3d_plugins/MIND/soma_segmentation.pro]: Is there a way for this value to not be hard coded here?

Comment (sigerma88) [reply released_plugins/v3d_plugins/MIND/soma_segmentation.pro]: We can probably reuse the environment variable `VAA3D_HOME` we created during the setup of Vaa3D

Comment (sigerma88) [reply released_plugins/v3d_plugins/MIND/soma_segmentation.pro]: or it seems like the other plugins use directly

`VAA3DPATH =   ../../../../v3d_external`

Comment (sjavaheri) [reply released_plugins/v3d_plugins/MIND/soma_segmentation.pro]: Ah yeah it came from VAA_3D home variable, but the generated code hard coded it. Good catch - I will fix it and merge

**Review by sigerma88 — APPROVED** [review]

**Pull request merged**